stdlib/seq.mli

```diff
@@ -15,26 +15,63 @@
 
 (* Module [Seq]: functional iterators *)
 
+(** {1 Functional Iterators} *)
+
+(** The type ['a t] is a {b delayed list}, i.e. a list where some evaluation
+    is needed to access the next element. This makes it possible to build
+    infinite sequences, to build sequences as we traverse them, and to transform
+    them in a lazy fashion rather than upfront.
+*)
+
 (** @since NEXT_RELEASE *)
 
-type +'a node =
+(** The type of delayed lists containing elements of type ['a].
+    Note that the concrete list node ['a node] is delayed under a closure,
+    not a [lazy] block, which means it might be recomputed every time
+    we access it. *)
```

Comment:
In the stdlib, doc comments come after declarations.  Could you follow this style?